trillian/ctfe/ct_server/main.go

```diff
@@ -113,23 +121,36 @@ func main() {
 		// Use a fixed endpoint resolution that just returns the addresses configured on the command line.
 		res = util.FixedBackendResolver{}
 	}
-	bal := grpc.RoundRobin(res)
-	conn, err := grpc.Dial(*rpcBackend, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithBalancer(bal))
-	if err != nil {
-		glog.Exitf("Could not connect to rpc server: %v", err)
+
+	// Dial all our log backends. If there's only one of them we use the
+	// blocking option as we can't serve until connected.
+	clientMap := make(map[string]trillian.TrillianLogClient)
+	for _, be := range beMap {
+		glog.Infof("Dialling backend: %v", be)
+		var conn *grpc.ClientConn
+		bal := grpc.RoundRobin(res)
+		if len(beMap) == 1 {
```

Comment:
It would probably be neater to build a slice of `grpc.DialOption`s and conditionally add to that:
```go
		bal := grpc.RoundRobin(res)
		opts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithBalancer(bal)}
		if len(beMap) == 1 {
			opts = append(opts, grpc.WithBlock())
		}
		conn, err := grpc.Dial(be.BackendSpec, opts...)
```

Reply:
Done.